claat/proto-renderer/html/render_test.go

```diff
@@ -0,0 +1,54 @@
+package html
+
+import (
+	"bytes"
+	"io"
+	"testing"
+
+	"github.com/googlecodelabs/tools/claat/proto-renderer"
+)
+
+// unsupportedType is a dummy type used to showcase the failures of rendering
+// non-proto types, aside from Basic types, since we take in "any" type
+// as rendering input.
+type unsupportedType struct{}
+
+func TestRender(t *testing.T) {
+	// TODO: update to be proto dependent on next PR
+	tests := []struct {
+		in  interface{}
+		out string
+		ok  bool
+	}{
+		{nil, "", false},
+		{"invalid input type", "", false},
+		{unsupportedType{}, "", false},
+		{genrenderer.NewSampleProtoTemplate("3"), "3", false},
+		{genrenderer.NewSampleProtoTemplate("3"), "", true},
+	}
+
+	for _, tc := range tests {
+		o, err := Render(tc.in)
+
+		if err != nil && tc.ok {
+			t.Errorf("\nRender(\n\t%#v\n)\nPanic: %v(false negative)\nWant: %#v", tc.in, err, tc.out)
+		}
+
+		if err == nil {
+			rndrOut := readerToString(o)
+			if !tc.ok {
+				t.Errorf("\nRender(\n\t%#v\n) = %#v\nWant error\n(false positive)", tc.in, rndrOut)
+			}
+			if tc.out != rndrOut {
+				t.Errorf("\nRender(\n\t%#v\n) = %#v\nWant: %#v", tc.in, tc.out, rndrOut)
+			}
+		}
+	}
+}
+
+// readerToString Making io.Reader more readable for errors
+func readerToString(i io.Reader) string {
+	var b bytes.Buffer
+	b.ReadFrom(i)
+	return b.String()
+}
```

Comment:
Errors here will be as follows:

![image](https://user-images.githubusercontent.com/4205668/59888478-f4eb9200-9395-11e9-8966-3131f102d8f0.png)